akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -458,16 +458,19 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
 
   private lazy val shutdownHook = new Thread {
     override def run(): Unit = {
-      if (hasBeenShutdown.compareAndSet(false, true)) {
+      if (!hasBeenShutdown.get) {
```

Comment:
It can be a race between ActorSystem terminate, which will call transport.shutdown and this jvm shutdown hook. Coordinated shutdown also has it's jvm shutdown hook. If coordinated shutdown process is already in progress this hook will wait on the latch, and run afterwards.
In the end we only want the actual transport shutdown to be done once, and therefore the compareAndSet on the `hasBeenShutdown`.